symphony-bdk-spring/symphony-bdk-app-spring-boot-starter/src/test/java/com/symphony/bdk/app/spring/auth/CircleOfTrustControllerTest.java

```diff
@@ -0,0 +1,176 @@
+package com.symphony.bdk.app.spring.auth;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.symphony.bdk.app.spring.SymphonyBdkAppProperties;
+import com.symphony.bdk.app.spring.auth.model.AppToken;
+import com.symphony.bdk.app.spring.auth.model.BdkAppError;
+import com.symphony.bdk.app.spring.auth.model.BdkAppErrorCode;
+import com.symphony.bdk.app.spring.auth.model.UserId;
+import com.symphony.bdk.app.spring.exception.GlobalControllerExceptionHandler;
+import com.symphony.bdk.app.spring.properties.AppAuthProperties;
+import com.symphony.bdk.core.auth.AppAuthSession;
+import com.symphony.bdk.core.auth.ExtensionAppAuthenticator;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.auth.jwt.UserClaim;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@RunWith(SpringRunner.class)
+public class CircleOfTrustControllerTest {
+
+  private MockMvc mockMvc;
+  private static final ObjectMapper MAPPER = new JsonMapper();
+
+  @MockBean SymphonyBdkAppProperties properties;
+  @MockBean ExtensionAppAuthenticator authenticator;
```

Comment:
Shouldn't be mocked. Take a look [here](https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/test/java/com/symphony/bdk/spring/SymphonyBdkMockedConfiguration.java)

Reply:
I think we only want to test the logic within the Controller, so IMHO, mocking the authenticator is enough?